src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,30 @@
+const { ApiError } = require('../exeptions/auth.error');
+const { jwtServices } = require('../services/jwt.services');
+const { usersServices } = require('../services/users.services');
+
+module.exports = {
+  authMiddleware: async (req, _, next) => {
+    const authorization = 'authorization';
+
+    const token =
+      (req.headers[authorization] &&
+        req.headers[authorization].split(' ')[1]) ||
+      req.cookies.token;
+
+    if (!token) {
+      throw ApiError.unauthorized('Authorization token is missing');
+    }
+
+    const { email } = jwtServices.verify(token, process.env.JW_REFRESH_KEY);
```

Comment:
Ensure that `process.env.JW_REFRESH_KEY` is correctly set in your environment variables. If this key is meant for access tokens, consider using a different key for refresh tokens to avoid confusion.